File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        dp = {} 

        def recur(day,own):
            if day == len( prices):
                return 0 
            if (day,own) in dp :
                return dp[(day,own)]
            result=recur(day+1,own)
            if own :
                result = max(result, prices[day] + recur(day+1,False))
            else : 
                result = max ( result, - prices[day] + recur(day+1,True))
            
            dp[(day,own)] = result
            return result 


        return recur(0,False)

File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int max=0 ; 

        for ( int i =0; i < heights.size() ; i++ ){
            for (int j = i ; j < heights.size() ; j++ ){
                if (heights[i]>=heights[j]){
                    if (max < heights[j]*(j-i)){
                        max=heights[j]*(j-i);
                    }
                }
                else {
                    if ( max < heights[i]*(j-i)){
                        max=heights[i]*(j-i);
                    }
                }
            }
        }
        return max;
    }
};
